content/en-us/reference/engine/classes/Debris.yaml

```
@@ -3,64 +3,61 @@ type: class
   method.
 
-  After the lifetime argument has elapsed (in seconds) the object is removed in
-  the same manner as `Class.Instance:Destroy()`.
-
   As Debris is a service it must be created using the
   `Class.ServiceProvider:GetService()` method.
 
-  **Why use Debris?**
+  #### Advantages
 
-  Beyond creating a bit of a mess, objects that are no longer required can use
-  up system memory and cause the game to run slower over time. For this reason
-  it is always advised to run the `Class.Instance:Destroy()` function on objects
-  you no longer need. However in many cases an object may have a specific period
-  of utility after which it needs to be destroyed.
+  Besides creating a bit of a mess, objects that are no longer required can use
+  up system memory and cause an experience to run slower over time. For this
+  reason, it's always advised to call `Class.Instance:Destroy()` on objects
+  you no longer need. In some cases, however, an object may have a specific
+  period of utility before it can be destroyed.
 
-  Take the example of projectile that has just been thrown. It could be cleaned
-  up using:
+  Consider a wall being smashed into individual bricks. If you want a
+  brick to linger for 3 seconds before being destroyed, you can use the
+  following code:
 
   ```lua
-  wait(3)
-  projectile:Destroy()
+  task.wait(3)
+  brick:Destroy()
   ```
 
-  However there are a number of issues with this approach. Firstly, it requires
-  yielding the code with a wait, which is not always desirable. Secondly, before
-  the 3 seconds have elapsed the object may have already been destroyed (for
-  example, if it reached `Class.Workspace.FallenPartsDestroyHeight`).
+  However, waiting causes the thread to yield which may not be desired. To avoid
+  yielding, a callback function can be scheduled to run on a new thread after 3
+  seconds:
 
   ```lua
-  delay(3, function()
-  	if projectile and projectile.Parent then
-  		projectile:Destroy()
-  	end
+  task.delay(3, function()
+  	brick:Destroy()
   end)
   ```
 
-  This solves the above issues, as it spawns a new thread to prevent the current
-  one from yielding and checks to see if it can be destroyed. However at this
-  point a simple command has already become quite complicated and an unnecessary
-  thread is being created.
+  While this solves the above issue by preventing the code from yielding, it
+  introduces a second drawback in that the scheduled callback will never run if
+  the script is disabled or destroyed before the callback runs.
 
-  This is where Debris comes in, and the following code addresses all of the
-  above issues.
+  This is where `Class.Debris` has a specific advantage, as it does not yield
+  the current thread and runs outside the context of the script, guaranteeing
+  the instance is eventually destroyed even if the script is disabled or
+  destroyed. The following code addresses both issues cited for
+  `Library.task.wait()` and `Library.task.delay()`:
```

```suggestion
This is where `Class.Debris` has a specific advantage, as it does not yield the current thread and it runs outside the context of the script, guaranteeing the instance is eventually destroyed even if the script is disabled or destroyed. The following code does not yield and gaurantees the instance will be destroyed:
```

`task.wait()` has both issues (yields + depends on calling thread to live)
`task.delay()` has only one issue (does not yield, but depends on calling thread to live)
`Debris:AddItem` has neither (does not yield and does not depend on calling thread to live)

They're not necessarily really "issues", just differences in behavior. Perhaps you do want it to yield, or perhaps you do want it to not be destroyed if the script is disabled.